packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -439,38 +439,40 @@ class _NavigationRailState extends State<NavigationRail> with TickerProviderStat
                 Expanded(
                   child: Align(
                     alignment: Alignment(0, groupAlignment),
-                    child: Column(
-                      mainAxisSize: MainAxisSize.min,
-                      children: <Widget>[
-                        for (int i = 0; i < widget.destinations.length; i += 1)
-                          _RailDestination(
-                            minWidth: minWidth,
-                            minExtendedWidth: minExtendedWidth,
-                            extendedTransitionAnimation: _extendedAnimation,
-                            selected: widget.selectedIndex == i,
-                            icon: widget.selectedIndex == i ? widget.destinations[i].selectedIcon : widget.destinations[i].icon,
-                            label: widget.destinations[i].label,
-                            destinationAnimation: _destinationAnimations[i],
-                            labelType: labelType,
-                            iconTheme: widget.selectedIndex == i ? selectedIconTheme : effectiveUnselectedIconTheme,
-                            labelTextStyle: widget.selectedIndex == i ? selectedLabelTextStyle : unselectedLabelTextStyle,
-                            padding: widget.destinations[i].padding,
-                            useIndicator: useIndicator,
-                            indicatorColor: useIndicator ? indicatorColor : null,
-                            indicatorShape: useIndicator ? indicatorShape : null,
-                            onTap: () {
-                              if (widget.onDestinationSelected != null) {
-                                widget.onDestinationSelected!(i);
-                              }
-                            },
-                            indexLabel: localizations.tabLabel(
-                              tabIndex: i + 1,
-                              tabCount: widget.destinations.length,
+                    child: SingleChildScrollView(
```

Comment:
@DattatreyaReddy Just a thought, but you could optimize this further by using a `CustomScrollView` with a `SliverList` and a `SliverChildBuilderDelegate`. Because we now have a scrollview, people could theoretically add infinite destinations.

Reply:
sure, I'll do that

Reply:
But, CustomScrollView is forcing the width to be infinite which throws Unbounded width error

Reply:
also using SliverList forces ScrollPhysics

Reply:
I don't know that a CustomScrollView is necessary, it looks like a simple ListView could work here, and be much more efficient when there are many items.

Reply:
@Piinks Ah yes, `ListView.builder()` should do indeed. Pardon my mistake.

Reply:
But @Piinks even using `ListView.builder()` will throws unbounded Width error

Reply:
Yes, it looks like it will need to be constrained. :)

It looks like the spec already defines constraints: https://m3.material.io/components/navigation-rail/specs#59102131-b6f0-4491-a4be-f2d72aba7c76

@TahaTesser do you know if this is already accounted for in the tokens?

![image](https://user-images.githubusercontent.com/16964204/221935693-e9f9eb1b-1b64-4a36-a365-ca2b1c1a89df.png)

Reply:
> @TahaTesser do you know if this is already accounted for in the tokens?

None that I know of.



cc: @Piinks 

I just investigated ListView with constrained box set to min width or extended min width. It looks like the width could be wider than minwidth if text scale factor is higher. This breaks a lot of existing tests. 
 
I'm not sure if the navigation rail suppose to get that wide and exceed min width when changing text scale factor.  (On Android, NavigationRail seems to be a cap min width and text size).
I think it is an existing issue, it also breaks indicator overlay when layout is unexpectedly wide. IMO, addressing the existing layout issues with text scale factor first then adding any scrollable functionality is better. I can take that and add it to my backlog.

Reply:
> IMO, addressing the existing layout issues with text scale factor first then adding any scrollable functionality is better.

SGTM. 

> I'm not sure if the navigation rail suppose to get that wide and exceed min width when changing text scale factor.

I am also not sure, maybe @HansMuller or @guidezpl will know?

Reply:
I did a quick test on Android after my comment. Even after maximising the text scale and display size in system settings NavigationRail on native doesn't get wide as Flutter one and text size is also capped. I will investigate this further and file an issue.

Reply:
Here is what I heard back:

```
The rail should not exceed the width constraints. Text wrapping to two lines is not ideal but ok.
There is also an option for a drawer that provides more text space.

Nav rail width should be constraint and not change ( unless change to expanded version).
This width constraint can also force designer or content writer to use concise and short title in the Nav.
```

Reply:
Thanks for getting this feedback. 

When I wrapped constrained box with list view to this current PR. 80 minWidth is easily exceeded with text scale factor 3.0, to be around 120 and adding such constraints pretty much fails every exisiting in the navigation_rail_test.dart. (these results aren't final, I will do more digging later)